frontend/app/routes/_gcweb-app.personal-information.mailing-address.edit.tsx

```diff
@@ -125,11 +132,48 @@ export default function PersonalInformationMailingAddressEdit() {
     <>
       {errorSummaryItems.length > 0 && <ErrorSummary id={errorSummaryId} errors={errorSummaryItems} />}
       <Form className="max-w-prose" method="post">
-        <InputField id="address" label={t('personal-information:mailing-address.edit.field.address')} name="address" required defaultValue={defaultValues.address} errorMessage={errorMessages.address} />
-        <InputField id="city" label={t('personal-information:mailing-address.edit.field.city')} name="city" required defaultValue={defaultValues.city} errorMessage={errorMessages.city} />
-        <InputField id="province" label={t('personal-information:mailing-address.edit.field.province')} name="province" defaultValue={defaultValues.province} errorMessage={errorMessages.province} />
-        <InputField id="postalCode" label={t('personal-information:mailing-address.edit.field.postal-code')} name="postalCode" defaultValue={defaultValues.postalCode} errorMessage={errorMessages.postalCode} />
-        <InputField id="country" label={t('personal-information:mailing-address.edit.field.country')} name="country" required defaultValue={defaultValues.country} errorMessage={errorMessages.country} />
+        <div className="checkbox gc-chckbxrdio">
+          <input id="copy-home-address" type="checkbox" name="ifCopyHomeAddress" checked={isChecked} onChange={checkHandler} />
+          <label id="copy-home-address" htmlFor="copy-home-address">
+            Copy home address
+          </label>
+        </div>
+
+        <InputField
+          id="address"
+          label={t('personal-information:mailing-address.edit.field.address')}
+          name="address"
+          required
+          key={isChecked ? homeAddressInfo?.address : addressInfo?.address}
+          defaultValue={defaultValues.address}
+          errorMessage={errorMessages.address}
+        />
+        <InputField id="city" label={t('personal-information:mailing-address.edit.field.city')} name="city" required key={isChecked ? homeAddressInfo?.city : addressInfo?.city} defaultValue={defaultValues.city} errorMessage={errorMessages.city} />
+        <InputField
+          id="province"
+          label={t('personal-information:mailing-address.edit.field.province')}
+          name="province"
+          key={isChecked ? homeAddressInfo?.province : addressInfo?.province}
```

Comment:
Resolving this conversation because I am wrong about how the `key` attribute can be used in React. See <https://react.dev/reference/react/useState#resetting-state-with-a-key>